lib/utils/constants.py

```diff
@@ -116,6 +118,17 @@ def __getitem__(self, name):
 )
 
 
+NodeSelectionType = (
+    Literal["all"] | Literal["principal"] | Literal["random"] | list[str]
```

Comment:
nit: I think `Literal` can be shortened. https://peps.python.org/pep-0586/#shortening-unions-of-literals

Reply:
Shouldn't the `list[str]` be `list[NodeSelection]`?

Reply:
Not currently. Maybe it the future we could so something like that